### PR TITLE
Ear 1147 sort comments order in older surveys

### PR DESCRIPTION
**Pull request by zimboflyman**
### What is the context of this PR?
https://collaborate2.ons.gov.uk/jira/browse/EAR-1147
In Prod - some of the older comments in questionnaires aren't being sorted correctly
Looks like all comments in the DB _before_ the migration to MongoDB - were going to the top
MongoDB saves dates as ISODate('2021-01-18T10:46:41.360Z')

The fix i've added is to convert dates toString() before sorting and doing the sorting in the front end as the replies were done.

This fix will need to be pushed to prod before we can really confirm the bug is ehem 'sorted' 😊

### How to review
make sure new comments and replies are sorting correctly locally

### CODE REVIEW

**Review by farres1 — DISMISSED** [review]

**Review by jasehumphr — APPROVED** [review]
Comments display as expected. Let's see how they do with some older ones

**Review by tjamesmac — CHANGES_REQUESTED** [review]
Do the changes need to be one the front-end? Can we just move the new sort to the resolvers? Otherwise we are sorting comments twice now

**Review by martyncolmer — APPROVED** [review]

File: eq-author/src/components/CommentsPanel/index.js [hunk 1]
```diff
@@ -402,8 +402,13 @@ const CommentsPanel = ({ componentId, me: { id: myId } }) => {
...
 
-  const displayComments = comments.map((item, index) => {
-    const { replies } = comments[index];
+  let sortedComments = comments.slice() || [];
+  sortedComments = sortedComments.sort((a, b) =>
+    b.createdTime.toString().localeCompare(a.createdTime.toString())
+  );
+
+  const displayComments = sortedComments.map((item, index) => {
+    const { replies } = sortedComments[index];
```
Comment (tjamesmac) [inline eq-author/src/components/CommentsPanel/index.js:411]:
```suggestion
    const { replies } = item;
```

**Review by tjamesmac — APPROVED** [review]
All good and the change is optional

**Review by farres1 — APPROVED** [review]

**Pull request merged**